test_utils/src/main.rs

```diff
@@ -178,6 +207,50 @@ fn main() -> Result<()> {
             let total_mint_shares = total_mint * usdn_divisor;
             print_int_u256_hex(total_mint_shares)?;
         }
+        Commands::ChainlinkPrice { feed_id, timestamp } => {
+            let chainlink_low_latency_api_key = std::env::var("CHAINLINK_DATA_STREAM_API_KEY")
+                .context("getting CHAINLINK_DATA_STREAM_API_KEY env variable")?;
+
+            let chainlink_user_secret = std::env::var("CHAINLINK_DATA_STREAM_API_SECRET")
+                .context("getting CHAINLINK_DATA_STREAM_API_SECRET env variable")?;
+
+            let chainlink_rest_url = std::env::var("CHAINLINK_DATA_STREAM_API_URL")
+                .context("getting CHAINLINK_DATA_STREAM_API_URL env variable")?;
+
+            const API_V1_REPORTS: &str = "/api/v1/reports";
+
+            let method = "GET";
+            let path = &format!("{API_V1_REPORTS}?feedID={feed_id}&timestamp={timestamp}");
+            let body = b"";
+            let client_id = &chainlink_low_latency_api_key;
+            let user_secret = &chainlink_user_secret;
+            let request_timestamp = SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .expect("Error: Timestamp in the past")
+                .as_millis();
+
+            let hmac_string = &generate_hmac(
```

Comment:
```suggestion
            let hmac_string = generate_hmac(
```

Same comment as above, you don't normally store references when defining variables, but add the `&` when you reference them later.